airflow/utils/log/file_task_handler.py

```diff
@@ -182,14 +184,38 @@ def set_context(self, ti: TaskInstance) -> None | SetContextPropagate:
         functionality is only used in unit testing.
 
         :param ti: task instance object
+        :param identifier: if set, adds suffix to log file. For use when shipping exceptional messages
+            to task logs from context other than task or trigger run
         """
-        local_loc = self._init_file(ti)
+        local_loc = self._init_file(ti, identifier=identifier)
         self.handler = NonCachingFileHandler(local_loc, encoding="utf-8")
         if self.formatter:
             self.handler.setFormatter(self.formatter)
         self.handler.setLevel(self.level)
         return SetContextPropagate.MAINTAIN_PROPAGATE if self.maintain_propagate else None
 
+    @cached_property
+    def _supports_arbitrary_ship(self):
+        return "identifier" in inspect.signature(self.set_context).parameters
+
+    def ship_arbitrary_message(
```

Comment:
do does this make more sense here or on the TaskLogShipper class???

Reply:
yes, moved there.